src/cpu/cpu_inner_product_list.cpp

```diff
@@ -83,6 +83,10 @@ const std::map<pk_dt_impl_key_t, std::vector<impl_list_item_t>> &impl_list_map()
             CPU_INSTANCE(ref_inner_product_fwd_t)
             nullptr,
         }},
+        {{forward, f32, bf16, f32}, {
```

Comment:
Could you, please, elaborate on why this configuration is needed?

Just in case, fpmath mode requires both src and weights to be fp32 to work. From functional point of view, the implementation takes care of down-converting and applying low precision computation.

Reply:
Hi @dzarukin , thanks for the review.  your explanation of the fastmath mode is what we have been using so far for the eager mode execution of the model, where we create primitive with both src and weights in fp32, and then reorder weights (from fp32 to bf16) for every exec. 
However, with graph compilation, we are able to reorder/pre-pack all the weights during the model load and compilation phase itself so that redundant and on-the-fly reorders can be avoided. This PR is to support the compile scenario where src in fp32 and weights in bf16 for the primitive create. ACL primitives already support this mode (i call it sbgemm), but lacking this definition in this list is preventing us from creating them. 
I will add some of these comments to the source file.